stdlib/socket/src/main/java/org/ballerinalang/stdlib/socket/endpoint/tcp/client/Close.java

```diff
@@ -68,6 +71,12 @@ public void execute(Context context) {
             if (client != null && Boolean.getBoolean(client.toString())) {
                 SelectorManager.getInstance().stop();
             }
+            SocketService socketService = (SocketService) clientEndpoint.getNativeData(SOCKET_SERVICE);
+            // Check whether client has the attached callback service.
+            // ResourceMap shouldn't be null if callback service attached.
+            if (socketService != null && socketService.getResources() != null) {
+                SelectorDispatcher.invokeOnClose(socketService);
```

Comment:
If it's null should we handle the error i.e log ?

Reply:
No. That's not an error situation. Having a callback service for a client is an optional.